server/src/test/java/io/crate/types/GeoShapeTypeTest.java

```diff
@@ -332,5 +332,35 @@ public void test_check_equality_of_shapes_of_same_type_with_different_coordinate
         assertThat(type.compare(shape1, shape2)).isEqualTo(1);
         assertThat(type.compare(shape2, shape1)).isEqualTo(1);
     }
+
+    @Test
+    public void test_geometry_collection_equal_to_corresponding_multi_part() throws Exception {
+        Map<String, Object> collection = parse(
+            """
+            {
+                "type": "GeometryCollection",
+                "geometries": [
+                    {
+                        "type": "Point",
+                        "coordinates": [1.0, 1.0]
+                    },
+                    {
+                        "type": "Point",
+                        "coordinates": [2.0, 2.0]
+                    }
+                ]
+            }
+            """
+        );
+        Map<String, Object> multiPoint = parse("""
+            {
+                "type": "MultiPoint",
+                "coordinates": [[1.0, 1.0], [2.0, 2.0]]
+            }
+            """
+        );
+        assertThat(type.compare(type.implicitCast(collection), type.implicitCast(multiPoint))).isEqualTo(0);
+        assertThat(type.compare(type.sanitizeValue(collection), type.sanitizeValue(multiPoint))).isEqualTo(0);
```

Comment:
Afaik in this cases, standard map equality checks should work, no need to call `type.compare()`.
```suggestion
        assertThat(type.implicitCast(collection)).isEqualTo(type.implicitCast(multiPoint));
        assertThat(type.sanitizeValue(collection)).isEqualTo(type.sanitizeValue(multiPoint));
```